plugins/suggest/suggest.c

```diff
@@ -657,10 +657,22 @@ learner_init_columns(grn_ctx *ctx, grn_suggest_learner *learner)
   grn_obj *seqs, *events, *post_item, *items, *pairs;
 
   learner->seqs = seqs = grn_ctx_at(ctx, GRN_OBJ_GET_DOMAIN(learner->seq));
+  if (!learner->seqs) {
+    ERR(GRN_DOMAIN_ERROR,
+        "failed to initialize columns because domain id: <%d> of seq is not found.",
+        GRN_OBJ_GET_DOMAIN(learner->seq));
+    return;
+  }
   learner->seqs_events = grn_obj_column(ctx, seqs, CONST_STR_LEN("events"));
 
   events_id = grn_obj_get_range(ctx, learner->seqs_events);
   learner->events = events = grn_ctx_at(ctx, events_id);
+  if (!learner->events) {
+    ERR(GRN_UNKNOWN_ERROR,
+        "failed to initialize columns because event id: <%d> is not found.",
```

Comment:
The last "." is needless.

Reply:
`: <...>` should be placed at the last. We should not put it the middle of sentence.

`"failed to refer sequence's event table: <%.*s>: <%u>(<%.*s>)", seqs_events_name_size, seqs_events_name, events_id, events_name_size, events_name)"`

`events_name` will be able to get by `grn_table_get_key()` against `grn_ctx_db()`.